TarSCM/archive.py

```diff
@@ -33,6 +33,9 @@ def extract_from_archive(self, repodir, files, outdir):
             if not os.path.exists(src):
                 sys.exit("%s: No such file or directory" % src)
 
+            if os.path.islink(src):
+                sys.exit("%s: Extracting symlinks is not permitted" % src)
```

Comment:
This check looks like it is insufficient when multiple components are specified like `mylink -> /etc`, `extract = "mylink/fstab"`. `os.path.islink` would return `False` although a symlink is involved. An approach using `realpath` or checking each component following `repodir` seems advisable.